test/tracer/tracer.c

```diff
@@ -0,0 +1,101 @@
+/*
+ * tracer.c
+ *
+ * RISC-V ISA: RV32I
+ *
+ * Copyright (C) 2017-2023 Tactical Computing Laboratories, LLC
+ * All Rights Reserved
+ * contact@tactcomplabs.com
+ *
+ * See LICENSE in the top level directory for licensing details
+ *
+ */
+
+#include "stdlib.h"
+#include "stdint.h"
+#include "rev-macros.h"
+
+#define assert TRACE_ASSERT
+
+// inefficient calculation of r-s
+int long_sub(int r, int s) {
+  for (int i=0;i<s;i++)
+    r--;
+  return r;
+}
+
+volatile int check_push_on(int x, int y) {
```

Comment:
`volatile` is not meaningful on function return values (but it could be important on the referents of references, or the pointees of pointers, of function return references or pointers, e.g., `volatile int& check_push_on(...)`